pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/internal/Helper.java

```diff
@@ -0,0 +1,230 @@
+        final ASTReferenceExpression ref = variable.getFirstChildOfType(ASTReferenceExpression.class);
+        String objectName = "";
+        if (ref != null) {
+            if (ref.getNode().getNames().size() == 1) {
+                objectName = ref.getNode().getNames().get(0).getValue() + ".";
+            }
+        }
+
+        VariableExpression n = variable.getNode();
+        StringBuilder sb = new StringBuilder().append(n.getDefiningType().getApexName()).append(":").append(objectName)
+                .append(n.getIdentifier().getValue());
+        return sb.toString();
+    }
+
+    public static String getFQVariableName(final ASTVariableDeclaration variable) {
+        VariableDeclaration n = variable.getNode();
+        StringBuilder sb = new StringBuilder().append(n.getDefiningType().getApexName()).append(":")
+                .append(n.getLocalInfo().getName());
+        return sb.toString();
+    }
+
+    public static String getFQVariableName(final ASTField variable) {
+        Field n = variable.getNode();
+        StringBuilder sb = new StringBuilder()
+                .append(n.getDefiningType().getApexName()).append(":")
+                .append(n.getFieldInfo().getName());
+        return sb.toString();
+    }
+
+    static String getVariableType(final ASTField variable) {
+        Field n = variable.getNode();
+        StringBuilder sb = new StringBuilder().append(n.getDefiningType().getApexName()).append(":")
+                .append(n.getFieldInfo().getName());
+        return sb.toString();
+    }
+    
+    public static String getFQVariableName(final ASTFieldDeclaration variable) {
+        StringBuilder sb = new StringBuilder()
+                .append(variable.getNode().getDefiningType().getApexName()).append(":")
+                .append(variable.getImage());
+        return sb.toString();
+    }
+
+    public static String getFQVariableName(final ASTNewKeyValueObjectExpression variable) {
+        StringBuilder sb = new StringBuilder()
+                .append(variable.getNode().getDefiningType().getApexName()).append(":")
+                .append(variable.getType());
+        return sb.toString();
+    }
+
+    public static boolean isSystemLevelClass(ASTUserClass node) {
+        List<TypeRef> interfaces = node.getNode().getDefiningType().getCodeUnitDetails().getInterfaceTypeRefs();
+
+        for (TypeRef intObject : interfaces) {
+            if (isWhitelisted(intObject.getNames())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean isWhitelisted(List<Identifier> ids) {
+        StringBuffer sb = new StringBuffer();
+
+        for (int i = 0; i < ids.size(); i++) {
+            sb.append(ids.get(i).getValue());
+
+            if (i != ids.size() - 1) {
+                sb.append(".");
+            }
+        }
+
+        switch (sb.toString().toLowerCase(Locale.ROOT)) {
+        case "queueable":
```

Comment:
Indentation?

Reply:
Hm... we seem to have a different (checkstyle) rule active for indenting switches... the way it is currently is correct (at least we enforce it so through checkstyle)